commit 497d9cdc67f0bdae929fcde677b5f441e94a6c8b
Author: Elias Hernandis <elias@hernandis.me>
Date:   Wed Feb 18 08:10:40 2026 +0100

    Fixed #36951 -- Removed empty exc_info from log_task_finished signal handler.
    
    Before, if no exception occurred, "None Type: None" was logged.

diff --git a/django/tasks/signals.py b/django/tasks/signals.py
index 288fe08e32..919dae0222 100644
--- a/django/tasks/signals.py
+++ b/django/tasks/signals.py
@@ -49,6 +49,8 @@ def log_task_started(sender, task_result, **kwargs):
 
 @receiver(task_finished)
 def log_task_finished(sender, task_result, **kwargs):
+    # Signal is sent inside exception handlers, so exc_info() is available.
+    exc_info = sys.exc_info()
     logger.log(
         (
             logging.ERROR
@@ -59,6 +61,5 @@ def log_task_finished(sender, task_result, **kwargs):
         task_result.id,
         task_result.task.module_path,
         task_result.status,
-        # Signal is sent inside exception handlers, so exc_info() is available.
-        exc_info=sys.exc_info(),
+        exc_info=exc_info if exc_info[0] else None,
     )
